Source/WebKit/UIProcess/UserMediaPermissionRequestManagerProxy.cpp

```diff
@@ -581,12 +578,18 @@ void UserMediaPermissionRequestManagerProxy::processUserMediaPermissionRequest()
             if (!m_page.hasRunningProcess() || !m_page.mainFrame())
                 return;
 
+            if (videoDevices.isEmpty() && request->requiresDisplayCapture() && m_page.preferences().mockCaptureDevicesEnabled())
+                videoDevices = WebCore::RealtimeMediaSourceCenter::singleton().displayCaptureFactory().displayCaptureDeviceManager().captureDevices();
```

Comment:
Why do we need this? Can we move this code to mock capture code if needed?
Maybe in processUserMediaPermissionValidRequest, around lines 651/654?

Reply:
We need this because when mock devices are active we don't necessarily go through the mock picker path. I will move it to `processUserMediaPermissionValidRequest`.